### ISSUE TITLE
Indent rule and nested one-line blocks

### ISSUE THREAD

**Issue by targos** [eslint/eslint#3238]
Rule:
```yml
indent: [2, 2, {"SwitchCase": 1, "VariableDeclarator": 2}]
```

Code:
```js
for (var val in obj)
  if (true)
    console.log(val);

if (true)
  if (true)
    if (true)
      console.log(true);
```

Error:
```
test4.js
  3:5  error  Expected indentation of 2 characters but found 4  indent
  7:5  error  Expected indentation of 2 characters but found 4  indent
  8:7  error  Expected indentation of 4 characters but found 6  indent
```

I expect no error in this case.

Cross-reference: https://github.com/nodejs/io.js/pull/2286

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!
